quickwit-core/src/indexing/statistics.rs

```diff
@@ -102,8 +112,6 @@ impl StatisticsCollector {
         let (event_sender, mut event_receiver) = channel(1000);
         task::spawn(async move {
             while let Some(event) = event_receiver.recv().await {
-                //TODO: discuss the grannular level of changes we want to display
-                // live in the terminal
                 let mut statistics = moved_statistics_collector.lock().await;
```

Comment:
there are only counters in there.
Should we remove use atomics and remove the mutex?

Or even better...
Remove the channel and use methods to increment instead of emitting events?